### PR TITLE
OU-200: Use korrel8r to load metrics related to logs

### PR DESCRIPTION
**Pull request by jgbernalp**
[OU-200](https://issues.redhat.com/browse/OU-200)

This PR adds a "Correlation" column in the logs table if korrel8r is reachable, when clicking on "metrics" a metric from the log entry is fetched from korrel8r and the console is redirected to metrics with the predefined query

https://github.com/openshift/logging-view-plugin/assets/5461414/36297671-22be-4842-898c-3a62b165b3ba

### CODE REVIEW

File: web/src/components/logs-table.tsx [hunk 1]
```diff
@@ -240,6 +250,87 @@ const LogRow: React.FC<LogRowProps> = ({ data, title, showResources }) => {
...
+
+                const metricsUrl = new URL(window.location.href);
+
+                const params = new URLSearchParams();
+                params.set('query0', promQL);
+                metricsUrl.pathname = '/monitoring/query-browser';
+                metricsUrl.search = params.toString();
+
+                // open the metrics page in a new tab
+                window.open(metricsUrl, '_blank');
```
Comment (kyoto) [inline web/src/components/logs-table.tsx:307]: Do we want to open in a new tab? I was thinking that we would redirect in the same tab.

File: web/src/components/logs-table.tsx [hunk 2]
```diff
@@ -240,6 +250,87 @@ const LogRow: React.FC<LogRowProps> = ({ data, title, showResources }) => {
...
+    }
+    if (namespace) {
+      queryLabels.push(`kubernetes_namespace_name="${namespace}"`);
+    }
+    if (pod) {
+      queryLabels.push(`kubernetes_pod_name="${pod}"`);
+    }
+
+    // We require a log type and at least one search label value to get related metrics
+    if (logType && queryLabels.length) {
```
Comment (kyoto) [inline web/src/components/logs-table.tsx:273]: If this condition is not satisfied, then `checkForMetrics` will do nothing, but there will be no UI feedback to explain why. In other words, I think that clicking the button will do nothing, but there will be nothing to explain why. Maybe the button should not even be shown if this condition is true?

File: web/src/components/logs-table.tsx (hunk 1 above)
Comment (jgbernalp) [reply web/src/components/logs-table.tsx:316]: Until we have the debugging panel for easy navigation, I think is valuable for users to keep the initial log list and click on result items for metrics that help troubleshoot

Comment (zhuje) [reply web/src/components/logs-table.tsx:316]: I agree, I think opening a new tab would be useful for navigating the logging/metrics pages we currently have.

**Review by zhuje — COMMENTED** [review]
/lgtm

**Pull request merged**